src/stores/video.ts

```diff
@@ -342,76 +461,130 @@ export const useVideoStore = defineStore('video', () => {
   const updateLastProcessingUpdate = (recordingHash: string): void => {
     const info = unprocessedVideos.value[recordingHash]
     info.dateLastProcessignUpdate = new Date()
-    unprocessedVideos.value = { ...unprocessedVideos.value, ...{ [recordingHash]: info } }
+    unprocessedVideos.value = {
+      ...unprocessedVideos.value,
+      ...{ [recordingHash]: info },
+    }
   }
 
-  const processVideoChunksAndTelemetry = async (recordingHash: string, info: UnprocessedVideoInfo): Promise<void> => {
-    if (info.dateFinish === undefined) {
-      info.dateFinish = info.dateLastRecordingUpdate
-    }
+  // Progress tracking for video processing
+  const videoProcessingDetails = ref<VideoProcessingDetails>({})
+  const totalFilesToProcess = ref(0)
 
-    // eslint-disable-next-line jsdoc/require-jsdoc
-    const chunks: { blob: Blob; name: string }[] = []
+  const updateFileProgress = (filename: string, progress: number, message: string): void => {
+    videoProcessingDetails.value[filename] = { filename, progress, message }
+  }
+
+  const currentFileProgress = computed(() => {
+    return Object.values(videoProcessingDetails.value).map((detail) => ({
+      fileName: detail.filename,
+      progress: detail.progress,
+      message: detail.message,
+    }))
+  })
+
+  const overallProgress = computed(() => {
+    const entries = Object.values(videoProcessingDetails.value)
+    const totalProgress = entries.reduce((acc, curr) => acc + curr.progress, 0)
+    return (totalProgress / (totalFilesToProcess.value * 100)) * 100
+  })
+
+  const debouncedUpdateFileProgress = useDebounceFn((filename: string, progress: number, message: string) => {
+    updateFileProgress(filename, progress, message)
+  }, 100)
```

Comment:
It would be better to have this method somewhere else. As it is right now, it is causing git to generate a false diff, as if the `processVideoChunksAndTelemetry` was removed, which is confusing git on finding the right diff in this entire method.